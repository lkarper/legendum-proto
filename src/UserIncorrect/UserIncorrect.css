.UserIncorrect__message {
    margin: 5px;
}