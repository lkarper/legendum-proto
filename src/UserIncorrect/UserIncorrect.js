import React from 'react';

const UserIncorrect = (props) => {
    const { page, userResponse } = props;

    let feedback;
    if (userResponse === page.incorrect_response_option_1) {
        feedback = page.response_if_incorrect_1;
    }
    if (userResponse === page.incorrect_response_option_2) {
        feedback = page.response_if_incorrect_2;
    }
    if (userResponse === page.incorrect_response_option_3) {
        feedback = page.response_if_incorrect_3;
    }

    return (
        <div
            role="alert"
        >
            <p>{feedback}</p>
            <p>Try again</p>
        </div>
    )
}

export default UserIncorrect;