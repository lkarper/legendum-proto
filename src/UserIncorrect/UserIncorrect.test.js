import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import UserIncorrect from './UserIncorrect';

describe('UserIncorrect component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<UserIncorrect />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<UserIncorrect />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
