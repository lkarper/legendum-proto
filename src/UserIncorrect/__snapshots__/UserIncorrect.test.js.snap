// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserIncorrect component renders the UI as expected 1`] = `
<UserIncorrect
  page={Object {}}
  userResponse=""
>
  <p>
    Error: Looks like something went wrong. Check your connection and try again.
  </p>
</UserIncorrect>
`;
